File: backend/src/models/POSItem.js
const mongoose = require('mongoose');

const posItemSchema = new mongoose.Schema({
  categoryKey: {
    type: String,
    required: true,
    uppercase: true,
    trim: true,
    enum: ['BUS', 'CANTEEN', 'VENDING_MACHINE']
  },
  name: {
    type: String,
    required: true,
    trim: true
  },
  description: {
    type: String,
    trim: true
  },
  price: {
    type: mongoose.Schema.Types.Decimal128,
    required: true,
    min: [0, 'Price must be positive'],
    set: val => parseFloat(val).toFixed(2),
    get: val => parseFloat(val.toString())
  },
  image: {
    type: String,
    trim: true
  },
  isAvailable: {
    type: Boolean,
    default: true
  },
  displayOrder: {
    type: Number,
    default: 0
  },
  metadata: {
    route: { type: String, trim: true },
    location: { type: String, trim: true },
    capacity: { type: Number },
    ingredients: [{ type: String, trim: true }]
  }
}, {
  timestamps: true,
  toJSON: { getters: true },
  toObject: { getters: true }
});

posItemSchema.index({ categoryKey: 1, displayOrder: 1 });
posItemSchema.index({ categoryKey: 1, isAvailable: 1 });

module.exports = mongoose.model('POSItem', posItemSchema);

File: mobile-app/nfc_app/lib/features/pos/domain/pos_item.dart
class POSItem {
  const POSItem({
    required this.id,
    required this.categoryKey,
    required this.name,
    this.description,
    required this.price,
    this.image,
    this.isAvailable = true,
    this.displayOrder = 0,
    this.metadata,
  });

  final String id;
  final String categoryKey;
  final String name;
  final String? description;
  final double price;
  final String? image;
  final bool isAvailable;
  final int displayOrder;
  final Map<String, dynamic>? metadata;

  factory POSItem.fromJson(Map<String, dynamic> json) {
    return POSItem(
      id: json['_id']?.toString() ?? json['id']?.toString() ?? '',
      categoryKey: json['categoryKey']?.toString() ?? '',
      name: json['name']?.toString() ?? '',
      description: json['description']?.toString(),
      price: (json['price'] as num?)?.toDouble() ?? 0,
      image: json['image']?.toString(),
      isAvailable: json['isAvailable'] as bool? ?? true,
      displayOrder: (json['displayOrder'] as num?)?.toInt() ?? 0,
      metadata: json['metadata'] as Map<String, dynamic>?,
    );
  }
}
